### PR TITLE
30 api search jdt

### PR DESCRIPTION
**Pull request by JulienDiTria**
api serach implements search such as:
- search cards by name
- search card by id
- serach cards by set code
- search set by set code

### CODE REVIEW

**Review by KelvinKappeler — CHANGES_REQUESTED** [review]
Thank you for your work. Don't forget to comment on the classes and methods including in the tests package so that we know more quickly the interest of certain features!

**Comment by KelvinKappeler** [comment]
I wanted to just add something else :
Why didn't you use the tools made available to you from the Magic API? For example, I see that you created a MagicCard in WebApi when it already exists in the magic API documentation: 

https://github.com/MagicTheGathering/mtg-sdk-kotlin/blob/develop/src/io/magicthegathering/kotlinsdk/model/card/MtgCard.kt

File: app/src/main/java/com/github/sdp/tarjetakuna/ui/webapi/WebApiFragment.kt [hunk 1]
```diff
@@ -43,20 +43,47 @@ class WebApiFragment : Fragment() {
...
+            hideKeyboard()
+            binding.apiResults.text = getString(R.string.api_waiting_results)
+            viewModel.getCardById(this.requireContext(), binding.cardId.text.toString())
+        }
+
         binding.apiSets.setOnClickListener {
+            hideKeyboard()
             binding.apiResults.text = getString(R.string.api_waiting_results)
             viewModel.getSets(this.requireContext())
         }
```
Comment (SolalPirelli) [inline app/src/main/java/com/github/sdp/tarjetakuna/ui/webapi/WebApiFragment.kt:73]: Looks like those listeners have a lot of common code that could be factored into one reusable function

File: app/src/test/java/com/github/sdp/tarjetakuna/ui/webapi/WebApiTest.kt [hunk 2]
```diff
@@ -15,25 +15,18 @@ class WebApiTest {
...
     @Test
-    fun test_getMagicApi() {
-        val api = WebApi.getMagicApi()
-        assert(api.getSets().isExecuted.not())
-        assert(api.getCards().isExecuted.not())
+    fun test_setUrl() {
+        val api = WebApi()
+        val url = "https://api.magicthegathering.io/v2/"
+        api.magicUrl = url
+        assert(api.magicUrl == url)
```
Comment (SolalPirelli) [inline app/src/test/java/com/github/sdp/tarjetakuna/ui/webapi/WebApiTest.kt:23]: What's the point of this test? And why does it not use JUnit asserts?

**Pull request merged**